kotlin-parser/src/test/java/org/archcnl/kotlinparser/parser/ModelExtractorTest.java

```diff
@@ -0,0 +1,90 @@
+package org.archcnl.kotlinparser.parser;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.archcnl.owlify.famix.codemodel.DefinedType;
+import org.archcnl.owlify.famix.codemodel.Project;
+import org.archcnl.owlify.famix.codemodel.SourceFile;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+public class ModelExtractorTest {
+
+  private final Path pathToPackage = Path.of("./src/test/resources/example-project/example/");
+  private final Path pathToPackageWithInvalidProject =
+      Path.of("./src/test/resources/example-project/example-invalid/");
+
+  private final String interfaceType = "Interface";
+  private final String getType = "GET";
+  private final String complexClass = "ComplexClass";
+  private final String classWithInnerClass = "ClassWithInnerClass";
+  private final String innerClass = "InnerClass";
+  private final String classInSubpackage = "ClassInSubpackage";
+  private final String anotherClass = "AnotherClass";
+
+  private final String baseNamespace = "example";
+
+  private ModelExtractor classUnderTest;
```

Comment:
The variable `classUnderTest` does not need to be in class scope but should be moved to the individual test methods.

Reply:
Moreover, the name is not very meaningful: Why not just call it `modelExtractor`?

Reply:
@toklinke it's about "matter of taste" in my opinion. You can see based on the type of a variable, what class is under test. But I adjusted it)

Reply:
:+1: